geopandas/sindex.py

```diff
@@ -70,6 +91,141 @@ def __init__(self, geometry):
                 # See https://github.com/Toblerity/rtree/issues/20.
                 super().__init__()
 
+            # store reference to geometries for predicate queries
+            self._geometries = geometry.geometry.values
+            # create a prepared geometry cache
+            self._prepared_geometries = np.array(
+                [None] * self._geometries.size, dtype=object
+            )
+
+        def query(self, geometry, predicate=None, sort=False):
+            """Compatibility layer for pygeos.query.
+
+            This is not a vectorized function, if speed is important,
+            please use PyGEOS.
+
+            Parameters
+            ----------
+            geometry : shapely geometry
+                A single shapely geometry to query against the spatial index.
+            predicate : {None, 'intersects', 'within', 'contains', 'overlaps', 'crosses', 'touches'}, optional
+                If predicate is provided, a prepared version of the input geometry is tested using
+                the predicate function against each item in the index whose extent intersects the
+                envelope of the input geometry: predicate(geometry, tree_geometry).
```

Comment:
This isn't technically correct, per below implementation.

```
If predicate is provided, the input geometry is tested using the predicate function against each item in the index whose extent intersects the envelope of the input geometry: predicate(geometry, tree_geometry).  If possible,
prepared geometries are used to help speed up the predicate operation.
```
(sorry for bad prior suggestion here)

Reply:
Thanks. I am going to leave any mention of prepared geometries out of the pygeos sindex docstrings since it currently does not use prepared geometries and if it does, that might happen internally within pygeos.